PeriodogramAnalysis/test_spectrum_analysis.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Optional, Tuple
+
+import pytest
+import numpy as np
+from spectrum_analysis import PowerSpectrumAnalysis
+from scipy.integrate import simpson
+
+
+def psd_input() -> dict[int, dict[str, Any]]:
+    freqs = np.array([1, 2, 3, 4, 5, 7, 9, 10, 15, 25, 35, 50])
+    psd = np.array(
+        [
+            [0.1, 0.2, 0.3, 0.4, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85],
+            [0.2, 0.25, 0.3, 0.35, 0.4, 0.425, 0.45, 0.475, 0.5, 0.525, 0.55, 0.575],
+        ]
+    ).T
+
+    output = (freqs, psd)
+
+    return output
+
+
+def test_call_invalid_input_parameters():
+    with pytest.raises(ValueError):
+        PowerSpectrumAnalysis(band_display=(5, 10, 20, 30))
+    with pytest.raises(ValueError):
+        PowerSpectrumAnalysis(band_display=(-2, 5))
+
+    with pytest.raises(ValueError):
+        PowerSpectrumAnalysis(band_display=((8, 5)))
+    with pytest.raises(ValueError):
+        PowerSpectrumAnalysis(band_display=((-3, 5)))
+    with pytest.raises(ValueError):
+        PowerSpectrumAnalysis(band_display=((8), (12, 30), (30, 100)))
+
+
+def test_call_return_shape():
+    analyzer = PowerSpectrumAnalysis()
+    result = analyzer(psd_input())
+    freqs, psd, psd_idx = result
+
+    assert isinstance(freqs, np.ndarray)
+    assert isinstance(psd, np.ndarray)
+    assert freqs.shape[0] == 12
+    assert psd.shape[0] == 12
+    assert psd_idx.shape == (12, 10)
+
+
+def test_computing_absolute_and_relative_power(mocker) -> None:
+    analysis = PowerSpectrumAnalysis()
+    result = analysis(psd_input())
+    freqs, psd, psd_idx = result
+
+    power_analysis = PowerSpectrumAnalysis()
+    mocker.patch.object(power_analysis, "num_channel", 2)
+
+    psd_idx, power, rel_power = power_analysis.computing_absolute_and_relative_power(
+        freqs, psd
+    )
+
+    for ch in range(2):
```

Comment:
I see you have two channels, but I don't see why. Just use 1 channel and check separately if multi-channel result can be handled properly.

Reply:
I changed the code here [9aaf6f8] but not sure if it is reasonable.